src/Lang/Cpp/CppDocGen.ts

```diff
@@ -228,6 +228,34 @@ export class CppDocGen implements IDocGen {
         }
     }
 
+    protected generateVersionTag(lines: string[]) {
+        if (this.cfg.File.versionTag.trim().length !== 0) {
+            lines.push(this.cfg.C.commentPrefix + this.cfg.File.versionTag);
+        }
+    }
+
+    protected generateCopyrightTag(lines: string[]) {
+        this.cfg.File.copyrightTag.forEach((element) => {
+            this.generateFromTemplate(
+                lines,
+                this.cfg.yearTemplateReplace,
+                element,
+                [moment().format("YYYY")],
+            );
+        });
+    }
+
+    protected generateCustomTag(lines: string[]) {
+        this.cfg.File.customTag.forEach((element) => {
+            this.generateFromTemplate(
+                lines,
+                this.cfg.dateTemplateReplace,
+                element,
+                [moment().format(this.cfg.Generic.dateFormat)],
```

Comment:
What about other tags like `@author` or `@file`?
Maybe this needs another switch case where different keywords can be handled.

Reply:
That's a great feature, I'll see if I can figure out how to  substitute multiple sub-strings within a template.  It's pretty common for us to use a "History" section in these headers, which would have lines like: ```* {date} {author} created file.```  (Have not used typescript before yesterday)

I'd love to figure out how to query the author name and e-mail from Git.

Reply:
You're doing an awesome job with typescript here then!
If you want to interact with Git maybe take a look at GitLens implementation:
https://github.com/eamodio/vscode-gitlens/blob/14e2aa67bfcd432a6b5af297cd5c20aafc182600/src/gitService.ts#L934
They have something to query the git user details for the current repo.
If you need some help you can feel free to ask!